docs/task_overview/plotting.md

```diff
@@ -0,0 +1,104 @@
+```{mermaid}
+:name: plotting-tasks
+:theme: forest
+:zoom:
+:align: center
+:caption: Class diagram of all plotting tasks
+classDiagram
```

Comment:
1.) Automatic generation would be nice, but it might be difficult to produce a similar result with that. In any way, I'd propose to try this in another PR.
regarding 2.) and 3.), these changes might be possible, but I did not manage to find out how to achieve this.